File: BFS/2606_바이러스/2606_바이러스.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
import java.util.StringTokenizer;
import java.util.List;
import java.util.LinkedList;
import java.util.Queue;

public class Main {
	public static void main(String args[]) throws IOException	{
    // set needed variables
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
    StringTokenizer st;
    Queue<Integer> q = new LinkedList<>();
    int N = Integer.parseInt(br.readLine());
    int M = Integer.parseInt(br.readLine());
    List<Edge>[] adjList = new List[N+1];
    for (int i = 1; i <= N; i++) adjList[i] = new LinkedList<>();
    boolean[] visited = new boolean[N+1];
    int infection = 0, from, to;

    // make graph
    for (int i = 0; i < M; i++){
      st = new StringTokenizer(br.readLine(), " ");
      from = Integer.parseInt(st.nextToken());
      to = Integer.parseInt(st.nextToken());
      adjList[from].add(new Edge(to));
      adjList[to].add(new Edge(from));
    }

    // do bfs
    q.add(1);
    visited[1] = true;
    while (!q.isEmpty()){
      int cur = q.poll();
      for (Edge e: adjList[cur]){
        if (!visited[e.adj]){
          q.add(e.adj);
          visited[e.adj] = true;
          infection++;
        }
      }
    }

    // print answer and end program
    System.out.println(infection);
    br.close();
    return;
  }
}

// Edge class
class Edge{
  int adj; // adjacent vertex number
  public Edge(int a){
    adj = a;
  }
}

File: BFS/2606_바이러스/2606_바이러스.py
# 2606번 BFS로 풀어보기

# 빠르게 입력받기
import sys
input = sys.stdin.readline

# import deque
from collections import deque

# 컴퓨터의 수를 N으로 받으세요
N = int(input())

# 연결되어있는 컴퓨터의 쌍의 수를 M으로 받으세요
M = int(input())

# graph list 선언하세요
graph = [[] for _ in range(N + 1)]

# visited list 선언하세요
visited = [False] * (N + 1)

# count 변수를 선언하세요
count = 0

# 연결되어있는 컴퓨터의 쌍을 입력받아서 그래프를 만드세요
for _ in range(M):
    node1, node2 = map(int, input().split())
    graph[node1].append(node2)
    graph[node2].append(node1)

# queue를 선언하세요
queue = deque()

# 1번 컴퓨터를 방문하세요
queue.append(1)
visited[1] = True

# bfs를 수행하세요
while queue:
    v = queue.popleft()
    count += 1
    for i in graph[v]:
        if not visited[i]:
            queue.append(i)
            visited[i] = True

# 결과를 출력하세요
print(count-1)